src/main/java/com/mineinabyss/mobzy/mobs/CustomEntity.kt

```diff
@@ -0,0 +1,85 @@
+package com.mineinabyss.mobzy.mobs
+
+import com.mineinabyss.geary.ecs.GearyEntity
+import com.mineinabyss.geary.ecs.components.addComponent
+import com.mineinabyss.geary.ecs.components.get
+import com.mineinabyss.geary.ecs.types.GearyEntityType
+import com.mineinabyss.geary.minecraft.components.BukkitEntityComponent
+import com.mineinabyss.geary.minecraft.store.BukkitEntityAccess
+import com.mineinabyss.geary.minecraft.store.decodeComponents
+import com.mineinabyss.idofront.events.call
+import com.mineinabyss.mobzy.api.nms.aliases.NMSEntity
+import com.mineinabyss.mobzy.api.nms.aliases.NMSSound
+import com.mineinabyss.mobzy.api.nms.aliases.toBukkit
+import com.mineinabyss.mobzy.ecs.components.ambient.Sounds
+import com.mineinabyss.mobzy.ecs.events.MobLoadEvent
+import com.mineinabyss.mobzy.registration.MobzyTypes
+import org.bukkit.SoundCategory
+import org.bukkit.persistence.PersistentDataContainer
+import org.bukkit.persistence.PersistentDataHolder
+import kotlin.random.Random
+
+/**
+ * A class for linking Bukkit entities to Geary entities. This is meant for Mobzy's own custom entities to use to
+ * share common code. We can't use an abstract class yet since we need our entities to extend existing NMS ones.
+ *
+ * @property gearyId The ID within Geary's ECS for this entity.
+ * This should always be implemented as `Engine.getNextId()`.
+ * @property nmsEntity The NMS equivalent of this geary entity.
+ * @property entity The bukkit equivalent of this geary entity.
+ *
+ * @see CustomMob
+ */
+interface CustomEntity : GearyEntity, PersistentDataHolder {
```

Comment:
Should all entities be PersistentDataHolders? I didn't see any specific use of the PDC in mobzy anyway, its mainly used in looty right? `encodeComponents` is never used in mobzy. I guess it can't hurt but maybe its not necessary.

Reply:
`Entity` already implements `PersitentDataHolder` so it makes sense all these custom bukkit entities should extend persisting data holder.